tests/e2e/utils/infrastructure.py

```diff
@@ -0,0 +1,197 @@
+import io
+import os
+import sys
+import zipfile
+from abc import ABC, abstractmethod
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List, Tuple, Type
+
+import boto3
+import yaml
+from aws_cdk import App, AssetStaging, BundlingOptions, CfnOutput, DockerImage, RemovalPolicy, Stack, aws_logs
+from aws_cdk.aws_lambda import Code, Function, LayerVersion, Runtime, Tracing
+
+PYTHON_RUNTIME_VERSION = f"V{''.join(map(str, sys.version_info[:2]))}"
+
+
+class PythonVersion(Enum):
+    V37 = {"runtime": Runtime.PYTHON_3_7, "image": Runtime.PYTHON_3_7.bundling_image.image}
+    V38 = {"runtime": Runtime.PYTHON_3_8, "image": Runtime.PYTHON_3_8.bundling_image.image}
+    V39 = {"runtime": Runtime.PYTHON_3_9, "image": Runtime.PYTHON_3_9.bundling_image.image}
+
+
+class InfrastructureStackInterface(ABC):
```

Comment:
```suggestion
class BaseInfrastructureStack(ABC):
```

for consistency across the code base.